deployments/common/config/all.yaml

```diff
@@ -6,6 +6,10 @@ jupyterhub:
     extraConfig:
       spawning: |
         c.KubeSpawner.start_timeout = 15 * 60  # Up to 15 minutes
+        c.KubeSpawner.args.append('--NotebookApp.shutdown_no_activity_timeout=14400') # Kills notebook pod if no kernels or terminals opened after 4 hours
```

Comment:
Good question. These were some numbers chosen more for testing than for production.

The reasoning behind the 1 hour to kill an idle kernel is to avoid memory loss when running a notebook and the user doesn't provide any input for 1 hour. I was thinking that giving 1 hour to kill an idle kernel may actually be low...Users may take a break or need to do some other research with other tools while using the notebooks... 

Regarding the notebook shutdown, I agree, 4 hours seems excessive. Maybe we can set the killing of idle kernels to 2 or 3 hours, and then shutdown the notebook server 30 minutes after no kernels. 

About the terminals opened, this will prevent culling. There seem to be options to also allow for culling with terminals opened, but that hasn't been tested yet.

Reply:
I'll add a little ticket to discuss and/or add terminal culling.

Reply:
I temporarily tweaked the values to:  

- 4 hours with no kernel activity kills the kernel
- 30 mins with no kernels or terminals running kills the notebook pod.